### ISSUE TITLE
Fix config.ini pull/push issues (25 cow points)

### ISSUE THREAD

**Issue by brifordwylie** [SuperCowPowers/old-workbench-archived#136]
The current 'approach' to deal with local changes to the config.ini files (one in server/ and one in clients/ ) sucks gigantic moth-balls. We have a section in the FAQ that talks about doing this:

```
git update-index --assume-unchanged workbench/server/config.ini
git update-index --assume-unchanged workbench/clients/config.ini
```

This is suboptimal for two reasons
- The user has to issue a complex git command
- I've seen situations where when trying to change branches git will tell you that you have uncommitted changes to these file and you can't swtich branches.

**Comment by jeffbaumes** [comment]
After a bit of investigation, there does not seem to be a better fix if you want to have single config files. My only thought is to git mv these to config.ini.example commit them, and .gitignore config.ini. Then at build time, or run time, check if config.ini exists and if not, copy config.ini.example to config.ini - or have the user manually perform the copy after a checkout. So the instructions move from those complex git commands to

```
cp workbench/server/config.ini.example workbench/server/config.ini
cp workbench/clients/config.ini.example workbench/clients/config.ini
```

A marginal improvement - this is a negative in a way since even non-developers would need to do this copy while only developers needed to do the update-index commands.

**Comment by brifordwylie** [comment]
Yeah... actually I used to have code that did exactly that.... default.ini
-> config.ini ... okay I'll do the copy... thx for looking into it :)

On Tue, Jul 1, 2014 at 8:14 AM, Jeffrey Baumes notifications@github.com
wrote:

> After a bit of investigation, there does not seem to be a better fix if
> you want to have single config files. My only thought is to git mv these to
> config.ini.example commit them, and .gitignore config.ini. Then at build
> time, or run time, check if config.ini exists and if not, copy
> config.ini.example to config.ini - or have the user manually perform the
> copy after a checkout. So the instructions move from those complex git
> commands to
> 
> cp workbench/server/config.ini.example workbench/server/config.ini
> cp workbench/clients/config.ini.example workbench/clients/config.ini
> 
> A marginal improvement - this is a negative in a way since even
> non-developers would need to do this copy while only developers needed to
> do the update-index commands.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/SuperCowPowers/workbench/issues/136#issuecomment-47660881
> .

## 

Brian Wylie
www.supercowpowers.com
Skype: briford.wylie
Phone: (505) 206-4295
